src/mod/irc.mod/chan.c

```diff
@@ -1335,6 +1335,61 @@ static int got354(char *from, char *msg)
   return 0;
 }
 
+/* got 353: NAMES
+ * <server> 353 <chan> :[+/@]nick [+/@]nick ....
+ *
+ * if userhost-in-names is enabled, nick is nick@userhost.com
+ * this function is added solely to handle userhost-in-names stuff, and will
+ * update hostnames for nicks received
+ */
+static int got353(char *from, char *msg)
+{
+  char *nameptr, *chname, *uhost, *nick, *p, *host;
+  struct chanset_t *chan;
+  int i;
+
+  if (find_capability("userhost-in-names")) {
+    newsplit(&msg);
+    newsplit(&msg); /* Get rid of =/@ prefix */
+    chname = newsplit(&msg);
+    nameptr = newsplit(&msg);
+    while ((uhost = newsplit(&nameptr))) {
+      if (!strcmp(uhost, "")) {
+        break;
+      }
+      fixcolon(uhost);
```

Comment:
I suggest moving fixcolon to before the loop as `fixcolon(nameptr);` - it only needs to remove the colon once at the beginning of the string if present ("x", ":x", ":x x x" are the options)

Reply:
done